roofit/roofitcore/src/RooGlobalFunc.cxx

```diff
@@ -222,13 +238,23 @@ namespace RooFit {
   RooCmdArg Label(const char* str) { return RooCmdArg("Label",0,0,0,0,str,0,0,0) ; }
   RooCmdArg Layout(Double_t xmin, Double_t xmax, Double_t ymin) { return RooCmdArg("Layout",Int_t(ymin*10000),0,xmin,xmax,0,0,0,0) ; }
   RooCmdArg Parameters(const RooArgSet& params) { return RooCmdArg("Parameters",0,0,0,0,0,0,&params,0) ; }
+  RooCmdArg Parameters(RooArgSet && params) { return Parameters(RooCmdArg::take(std::move(params))) ; }
   RooCmdArg ShowConstants(Bool_t flag) { return RooCmdArg("ShowConstants",flag,0,0,0,0,0,0,0) ; }
 
   // RooTreeData::statOn arguments
   RooCmdArg What(const char* str) { return RooCmdArg("What",0,0,0,0,str,0,0,0) ; }
 
   // RooProdPdf::ctor arguments
   RooCmdArg Conditional(const RooArgSet& pdfSet, const RooArgSet& depSet, Bool_t depsAreCond) { return RooCmdArg("Conditional",depsAreCond,0,0,0,0,0,0,0,0,0,&pdfSet,&depSet) ; } ;
+  RooCmdArg Conditional(RooArgSet && pdfSet, const RooArgSet& depSet, Bool_t depsAreCond) {
+    return Conditional(RooCmdArg::take(std::move(pdfSet)), depSet, depsAreCond);
+  }
+  RooCmdArg Conditional(const RooArgSet& pdfSet, RooArgSet && depSet, Bool_t depsAreCond) {
+    return Conditional(pdfSet, RooCmdArg::take(std::move(depSet)), depsAreCond);
+  }
+  RooCmdArg Conditional(RooArgSet && pdfSet, RooArgSet && depSet, Bool_t depsAreCond) {
+    return Conditional(RooCmdArg::take(std::move(pdfSet)), RooCmdArg::take(std::move(depSet)), depsAreCond);
+  }
```

Comment:
Did you think about
```c++
template<Set RooArgSet_t>
RooCmdArg Conditional(RooArgSet_t&& pdfSet, ...) {
  return Conditional(std::forward<RooArgSet_t>(pdfSet), ...);
  }
  ```
  This way, it could collapse into a single constructor, and it would also take a `RooArgList`. I'm not sure if one template function is better than three overloads, but I wanted to put it up as an alternative.

Reply:
Yes, I was thinking about this! The problem here is that it doesn't work with Python. We are already lucky that cppyy correctly chooses `foo(T const&)` or `foo(T &&)`, but unfortunately it doesn't choose the right template instantiation when you choose universal references :(

Reply:
It would work with one templated function that is used by two non-templated functions, but that's again more verbose...